### PR TITLE
Updated opendds_idl code generation for unions

### PR DESCRIPTION
**Pull request by mitza-oci**
Reduce warnings in generated code.
Other minor cleanup.

### CODE REVIEW

File: dds/idl/dds_generator.h [hunk 1]
```diff
@@ -510,47 +506,43 @@ inline void generateCaseBody(CommonFn commonFn,
...
       rhs = "tmp.out()";
     } else if (br_cls & CL_ARRAY) {
-      forany = "      " + brType + "_forany fa = tmp;\n";
+      forany = "    " + brType + "_forany fa = tmp;\n";
       rhs = getWrapper("fa", br, WD_INPUT);
     } else {
       rhs = getWrapper("tmp", br, WD_INPUT);
     }
     be_global->impl_ <<
-      "    {\n"
```
Comment (simpsont-oci) [inline dds/idl/dds_generator.h]: Was it intentional to remove the level of scope provided by curly braces here? If you have multiple cases like this, won't you now have two instances of "tmp" in the same scope and get those warnings about "jump to case label crosses initialization of XXX?"

File: dds/idl/langmap_generator.cpp [hunk 2]
```diff
@@ -388,16 +388,16 @@ class GeneratorBase
...
-        "      this->_u." << name << " = other._u." << name << ";\n";
+        "    this->_u." << name << " = other._u." << name << ";\n";
     } else if (cls & CL_STRING) {
       ss <<
-        "      this->_u." << name << " = (other._u." << name << ") ? ::CORBA::string_dup(other._u." << name << ") : 0 ;\n";
+        "    this->_u." << name << " = (other._u." << name << ") ? ::CORBA::string_dup(other._u." << name << ") : 0 ;\n";
     } else if (cls & CL_ARRAY) {
       ss <<
-        "    this->_u." << name << " = (other._u." << name << ") ? " << map_type(field_type) << "_dup(other._u." << name << ") : 0 ;\n";
+        "  this->_u." << name << " = (other._u." << name << ") ? " << map_type(field_type) << "_dup(other._u." << name << ") : 0 ;\n";
```
Comment (simpsont-oci) [inline dds/idl/langmap_generator.cpp]: Does the array get indented less than the others, or is this a typo?

Comment (mitza-oci) [reply dds/idl/langmap_generator.cpp]: Probably a bug, looks like we need coverage for the case where an array is directly in a union.  I'll add...

File: tests/DCPS/Presentation/main.cpp [hunk 3]
```diff
@@ -270,14 +271,14 @@ ACE_TMAIN(int argc, ACE_TCHAR** argv)
         {
           ACE_ERROR_RETURN((LM_ERROR,
                             ACE_TEXT("%N:%l main()")
-                            ACE_TEXT(" ERROR: Unexpected number of samples taken!\n")), -1);
+                            ACE_TEXT(" ERROR: take failed %d\n"), error), -1);
```
[the remark this answers is not in the document]
Comment (simpsont-oci) [reply tests/DCPS/Presentation/main.cpp]: Gotcha. Sorry, I see that now. Still getting used to how Github displays things.

File: dds/idl/langmap_generator.cpp (hunk 2 above)
Comment (mitza-oci) [reply dds/idl/langmap_generator.cpp]: It's covered in tests/DCPS/Compiler/FooDef.idl (BigUnion).  I'll push a change.

**Pull request merged**